### PR TITLE
[API Server] Service Instance Watch API

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apiserver/internal/v1/instance.go [hunk 1]
```diff
@@ -160,6 +154,69 @@ func (o instanceServer) Delete(ctx context.Context, req *v1.InstanceDeleteReques
...
+			any, err := ptypes.MarshalAny(instance)
+			if err != nil {
+				return status.Errorf(codes.Internal, "marshalling instance to Any: %s", err.Error())
+			}
+			err = stream.Send(&v1.WatchEvent{
+				Type:   string(event.Type),
+				Object: any,
+			})
+			if grpcStatus, ok := err.(toGRPCStatus); ok {
+				return status.Error(grpcStatus.GRPCStatus().Code(), grpcStatus.GRPCStatus().Message())
```
Comment (aborilov) [inline pkg/apiserver/internal/v1/instance.go:193]: I think you can call `grpcStatus.Err()`

Comment (jiachengxu) [reply pkg/apiserver/internal/v1/instance.go]: Thanks, fixed

File: test/integration/apiserver.go [hunk 2]
```diff
@@ -216,6 +216,65 @@ func newAPIServer(f *testutil.Framework) func(t *testing.T) {
 	}
 }
 
+func fetchUserToken(ctx context.Context, t *testing.T, managementClient *testutil.RecordingClient, kubeconfig string) string {
```
[the remark this answers is not in the document]
Comment (jiachengxu) [reply test/integration/apiserver.go:219]: This is not polluting the PR, if you check the apiserver.go, you can see the `fecthUserToken` is mixing with our services tests functions, I would like to move the instanceService function lower and put our services tests functions together. Just no need to open a dedicated PR just for restructure things

File: test/integration/apiserver.go [hunk 3]
```diff
@@ -764,7 +788,6 @@ func offeringService(ctx context.Context, conn *grpc.ClientConn, managementClien
 		}, offeringCtx.Done()))
 
 		// watch offerings
-		t.Cleanup(cancel)
```
[the remark this answers is not in the document]
Comment (jiachengxu) [reply test/integration/apiserver.go:767]: This is not needed, and just can be removed. Also, don't think it is need to open a dedicated PR for it

File: test/integration/apiserver.go [hunk 4]
```diff
@@ -216,6 +216,65 @@ func newAPIServer(f *testutil.Framework) func(t *testing.T) {
...
+		},
+		fmt.Sprintf("https://localhost:%d/auth", localDexServerPort),
+		"admin@example.com",
+		"password",
+	)
+	require.NoError(t, err, "getting token from internal dex instance")
+	return token
+}
+
+type toGRPCStatus interface {
```
[the remark this answers is not in the document]
Comment (jiachengxu) [reply test/integration/apiserver.go:258]: done

File: pkg/apiserver/internal/v1/instance.go [hunk 5]
```diff
@@ -166,6 +160,69 @@ func (o instanceServer) Delete(ctx context.Context, req *v1.InstanceDeleteReques
...
+			}
+			any, err := ptypes.MarshalAny(instance)
+			if err != nil {
+				return status.Errorf(codes.Internal, "marshalling instance to Any: %s", err.Error())
+			}
+			err = stream.Send(&v1.WatchEvent{
+				Type:   string(event.Type),
+				Object: any,
+			})
+			if grpcStatus, ok := err.(toGRPCStatus); ok {
```
[the remark this answers is not in the document]
Comment (jiachengxu) [reply pkg/apiserver/internal/v1/instance.go]: done

**Pull request merged**